projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -1391,9 +1443,21 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     public onFilteringDone = new EventEmitter<IFilteringExpressionsTree>();
 
     /**
-     * Emitted when paging is performed.
+     * Emitted before paging is performed.
+     * @remarks
+     * Returns an object consisting of the previous and current pages.
+     * @example
+     * ```html
+     * <igx-grid #grid [data]="localData" [height]="'305px'" [autoGenerate]="true" (onPagingDone)="pagingDone($event)"></igx-grid>
+     * ```
+     */
+    @Output()
+    public onPaging = new EventEmitter<IPagingEventArgs>();
```

Comment:
The event is with on prefix because of the existing paging method?

Reply:
Exactly. I think it is still a matter of change in the naming, if we decide to.

Reply:
Probably it could happen in that mass event naming change